elasticdl/python/master/task_dispatcher.py

```diff
@@ -131,9 +132,19 @@ def create_tasks(self, task_type, model_version=-1):
             self._todo.extend(tasks)
         else:
             with self._lock:
-                self._todo.extend(tasks)
+                self._eval_todo.extend(tasks)
```

Comment:
I also don't know the reason for the difference. But, I think we also need a lock to operate `_todo`.

Reply:
Okay, we can take a look at this lock later after this PR.

Reply:
We cannot use lock here because the lock has been used by `get` or `get_eval_task` in TaskDispatcher. A deadlock will happen if the lock is used for `_todo`. And I think we can remove the lock for `_eval_to_do` too.